laws-africa/indigo
# Profile photos

#### Description
This PR adds profile pictures to Indigo. User will be able to upload, change and remove profile pictures that will appear in select areas of the platform.

Deliverables:
- [x] Profile picture visible on profile page
- [x] Remove/Edit profile picture functionality on profile edit page
- [x] Import Google profile pictures upon sign-up using Google
- [x] Migration to import photos of existing users with connected Google accounts
- [x] Thumbnails on the leaderboard on the Overview Page
- [x] Thumbnails on the user popup used across Indigo
- [x] Confirmation popup when removing current profile picture
- [x] Post processing uploaded images i.e resizing, cropping
- [x] Caching of profile pictures

#### Manual Testing
- A default avatar is displayed for users without profile pictures and should be visible on the user profile page, user profile edit page, user popup. Navigating to any of these pages should display the default user avatar or their custom avatar if one is available.
- You should be able to upload a new profile picture or remove an existing one in the edit profile section.
- Signing up a new user using Google should import their profile photo, save it, and display it as their profile photo across Indigo.

#### Relevant Issue
closes #456 

#### Context
- Wording on the disclaimer on edit/upload section subject to review
- Default profile picture is a fontawesome free icon in SVG format

#### Screenshots

User Profile page with a profile photo:
![Screenshot_2020-04-14 Robley Adrian – Indigo Platform](https://user-images.githubusercontent.com/8082197/79330457-3ef3c480-7f22-11ea-8555-725a8539d219.png)

User profile using the default profile picture:
![image](https://user-images.githubusercontent.com/8082197/79330514-5af76600-7f22-11ea-89af-d282cdc2e3a4.png)

Profile edit section for a user without a profile photo:
![image](https://user-images.githubusercontent.com/8082197/79330612-80846f80-7f22-11ea-821a-57357ec6bc5e.png)

Profile edit section for a user changing an existing profile photo:
![image](https://user-images.githubusercontent.com/8082197/79330921-056f8900-7f23-11ea-988c-a81cb2f2a162.png)

Leaderboard with thumbnails:
![image](https://user-images.githubusercontent.com/8082197/79346817-77ec6300-7f3b-11ea-84bd-140ad875da75.png)


User Popup with thumbnail:
![Screenshot from 2020-04-15 16-57-46](https://user-images.githubusercontent.com/8082197/79345918-68204f00-7f3a-11ea-94b9-b4cc225a3075.png)
